bin/plotting/pycbc_plot_bank_corner

```diff
@@ -0,0 +1,286 @@
+    for p in bankf.keys():
+        if not isinstance(bankf[p], h5py.Dataset): continue
+        # Ignore things which aren't numbers, which cannot
+        # be histogrammed - e.g. approximant
+        if not np.issubdtype(bankf[p].dtype, np.number): continue
+        bank[p] = bankf[p][:]
+        banklen = bankf[p].size
+
+logging.info("Got %d templates from the bank", banklen)
+
+# If no parameters are given - just plot whatever is in the bank
+if not args.parameters:
+    # Plot anything which is not singular
+    args.parameters = [k for k in bank
+                       if len(np.unique(bank[k])) > 1]
+    args.parameters_labels = {k: k for k in bank}
+
+# Check through the conversion options to see if any are used in
+# the parameters argument:
+cpar = args.color_parameter[0] if args.color_parameter else False
+cpar_label = args.color_parameter_labels[cpar] if cpar else False
+
+required_params = []
+for co in conversion_options:
+    if any([co in par for par in args.parameters]):
+        required_params.append(co)
+    if cpar and co in cpar:
+        required_params.append(co)
+
+# Check for possible double-counting of duration parameter:
+if 'duration' in required_params and any([p.endswith('_duration')
+                                          for p in required_params]):
+    # Check whether duration was specified directly:
+    dur_pars = [par for par in args.parameters if 'duration' in par]
+    _dur_pars = [par for par in args.parameters if '_duration' in par]
+    if len(dur_pars) == len(_dur_pars):
+        # It looks like duration has been added where it didnt need to be
+        required_params.remove('duration')
+
+logging.info("Required parameters to get from bank: %s",
+             ', '.join(required_params))
+
+# Do the same with the duration functions, but here we need to make
+# sure we have some other keys in order to calculate duration
+duration_required_keys = ['mass1', 'mass2', 'spin1z',
+                          'spin2z', 'f_lower']
+
+if any(['duration' in par for par in args.parameters]):
+    required_params += duration_required_keys
+
+# Some things may have been double counted - undo this
+required_params = np.unique(required_params)
+
+# Get parameters not directly in the bank:
+for p in required_params:
+    if p in bank: continue
+    if p not in bconv.conversion_options:
+        raise KeyError(f"ERROR Parameter {p} not in bank or conversion "
+                       "options, choose from bank parameters or " + \
+                       ', '.join(bconv.conversion_options))
+
+    logging.info("Converting %s", p)
+    bank[p] = bconv.get_bank_property(p, bank, np.arange(banklen))
+
+# Durations can be negative where get_imr_duration is not perfect - 
+# set a minimum value of 0.15s
+if 'duration' in bank:
+    bank['duration'] = np.maximum(bank['duration'], 0.15)
+if 'template_duration' in bank:
+    bank['template_duration'] = np.maximum(bank['template_duration'], 0.15)
```

Comment:
So is there no way of plotting durations shorter than 150 ms, even if computed correctly? I would say that a plotting code should just plot whatever comes out of the duration calculation…

Reply:
This was just a way of getting around the negative duration issue, and was valid for the purpose I used it for, then forgot to fix it. I'm not sure what is best to say to do for this though

Reply:
Treat the duration as any other value. If the user tries to use a log duration scale, and some durations are negative, the code will produce an error. This seems a perfectly reasonable behavior to me.